command/agent/agent.go

```diff
@@ -127,9 +127,29 @@ func (a *Agent) serverConfig() (*nomad.Config, error) {
 	}
 	if addr := a.config.Addresses.RPC; addr != "" {
 		conf.RPCAddr.IP = net.ParseIP(addr)
+	} else if device := a.config.Interfaces.RPC; device != "" {
+		ip, err := ipOfDevice(device)
+		if err != nil {
+			return nil, err
+		}
+		conf.RPCAddr.IP = ip
 	}
 	if addr := a.config.Addresses.Serf; addr != "" {
 		conf.SerfConfig.MemberlistConfig.BindAddr = addr
+	} else if device := a.config.Interfaces.Serf; device != "" {
+		ip, err := ipOfDevice(device)
+		if err != nil {
+			return nil, err
+		}
+		conf.SerfConfig.MemberlistConfig.BindAddr = ip.String()
+	}
+
+	if device := a.config.Interfaces.HTTP; device != "" {
```

Comment:
@dadgar If Addresses.HTTP is present and device is not present, then Nomad will use the Addresses.HTTP. Look at NewHTTPServer. However, I am wondering what we should do if by chance the operator specifies both the Addresses.HTTP and Interfaces.HTTP? I think we should just use Addresses.HTTP.